### DISCUSSION TITLE
One Script used by multiple engines

### DISCUSSION THREAD

**Discussion by diegodemarco** [sebastienros/jint#1302] [Q&A] [answered]
Hi!

To improve the performance of an Asp.Net application that uses scripting extensively, I would like to cache the Script objects instead of the script texts, which is what we're doing now.
Then, in each request, a new engine is created, and a function is executed, with parameters that if course vary from one request to another.
Note that each request is handled in its own thread, which should not be a problem because each request uses its own Engine. However, at any given time, two or more engines may be using the same (cached) Script instance.

In short, is there any problem if the same Script instance is used to feed separate engines, each running in its own thread?

Thank you!
Cheers, Diego

**Comment by lahma** [comment, marked as answer]
It's safe to share `Script` instance between engines, this is specifically known scenario and thus supported. Ideally you should use `Engine.PrepareScript` (available in latest beta) to prepare the script, it will calculate things that would otherwise be calculated runtime and causing unnecessary overhead.

**Reply by diegodemarco** [reply]
Thank you, your detailed explanation is very much appreciated!
